kolena/workflow/test_case.py

```diff
@@ -141,6 +293,7 @@ def _create_from_data(cls, data: CoreAPI.EntityData) -> "TestCase":
         obj._freeze()
         return obj
 
+    @check_edit
     def _hydrate(self, test_samples: List[Tuple[TestSample, GroundTruth]], description: Optional[str] = None) -> None:
         if len(test_samples) == 0:
             log.warn("no test samples provided, unable to populate test case")
```

Comment:
How will `create` and `load` work within a bulk editor context?

Reply:
Should behave the same as single test-case `__init__`. Any specifics you're asking?

Reply:
If you're asking about class methods `create` and `load`, currently they are not included in bulk processing, and therefore will proceed as normal.

Reply:
Given all of the constraints here, I'm wondering if using the same `TestCase` definition is still worth it:

- Must exit context before trying to create test suite
- Can't use `create` or `load` in context
- Weirdness around version 0 and version 1
- Uncertainty about how `reset` operates

Would it ultimately be cleaner to expose an API like this?

```python3
TS = List[Tuple[TestSample, GroundTruth]]

class TestCase:
    ...
    def bulk_create(
        test_cases: List[Tuple[str, List[Tuple[TestSample, GroundTruth]]]],
        reset: bool = False,  # throw if any test cases already exist, unless reset=True (overwrite)
    ) -> List[TestCase]:
        ...
```

What do you think?

Reply:
I like the current API because it’s easy to use.  You just need to remember to add the context `with TestCase.bulk_process():` to utilize the bulk processing.

> Must exit context before trying to create test suite

I personally don’t have any issue with this.  Since the context manager name includes TestCase, I think it makes sense to only create / edit test cases.

> Can’t use create or load in context

I think being able to support this in context is important.  I’m sure some users are still using create and load

> Weirdness around version 0 and version 1

I didn’t understand this part so can’t comment on this.

> Uncertainty about how `reset` operates

Do we know what the uncertainty here is?  Having a `reset` option is crucial when implementing the seeding script.